packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -187,14 +191,40 @@ const App = (props) => {
                 body: {}
             })
         }
+    }, [baskets])
+
+    useEffect(() => {
         // update the basket currency if it doesn't match the current locale currency
-        if (baskets?.baskets?.[0]?.currency && baskets.baskets[0].currency !== currency) {
+        if (basket?.currency && basket?.currency !== currency) {
             updateBasket.mutate({
-                parameters: {basketId: baskets.baskets[0].basketId},
+                parameters: {basketId: basket.basketId},
                 body: {currency}
             })
         }
-    }, [baskets])
+
+        // update the basket customer email
+        if (
+            basket &&
+            customer?.isRegistered &&
+            customer?.customerId === basket?.customerInfo?.customerId &&
+            customer?.email &&
+            customer?.email !== basket?.customerInfo?.email
+        ) {
+            updateCustomerForBasket.mutate({
+                parameters: {basketId: basket.basketId},
+                body: {
+                    email: customer?.email
```

Comment:
email should always be available at this step now.

Reply:
The `?` has been removed on this line